src/network/networkd-wwan-bus.c

```diff
@@ -0,0 +1,364 @@
+                { "address", "s", map_in6, offsetof(Bearer, ip6_address)   },
+                { "prefix",  "u", NULL,    offsetof(Bearer, ip6_prefixlen) },
+                { "dns1",    "s", map_dns, 0,                              },
+                { "dns2",    "s", map_dns, 0,                              },
+                { "dns3",    "s", map_dns, 0,                              },
+                { "gateway", "s", map_in6, offsetof(Bearer, ip6_gateway)   },
+                { "mtu",     "u", NULL,    offsetof(Bearer, ip6_mtu)       },
+                {}
+        };
+
+        return bus_message_map_all_properties(m, map, 0, error, userdata);
+}
+
+static int map_ip_type(sd_bus *bus, const char *member, sd_bus_message *m, sd_bus_error *error, void *userdata) {
+        AddressFamily *ip_type = ASSERT_PTR(userdata);
+        unsigned u;
+        int r;
+
+        assert(m);
+
+        r = sd_bus_message_read_basic(m, 'u', &u);
+        if (r < 0)
+                return r;
+
+        switch (u) {
+        case MM_BEARER_IP_FAMILY_NONE:
+                *ip_type = ADDRESS_FAMILY_NO;
+                break;
+        case MM_BEARER_IP_FAMILY_IPV4:
+                *ip_type = ADDRESS_FAMILY_IPV4;
+                break;
+        case MM_BEARER_IP_FAMILY_IPV6:
+                *ip_type = ADDRESS_FAMILY_IPV6;
+                break;
+        case MM_BEARER_IP_FAMILY_IPV4V6:
+                *ip_type = ADDRESS_FAMILY_YES;
+                break;
+        }
+
+        return 0;
+}
+
+static int map_properties(sd_bus *bus, const char *member, sd_bus_message *m, sd_bus_error *error, void *userdata) {
+        static const struct bus_properties_map map[] = {
+                { "apn",     "s", NULL,        offsetof(Bearer, apn)     },
+                { "ip-type", "u", map_ip_type, offsetof(Bearer, ip_type) },
+                {}
+        };
+
+        return bus_message_map_all_properties(m, map, BUS_MAP_STRDUP, error, userdata);
+}
+
+static int bearer_get_all_handler(sd_bus_message *message, void *userdata, sd_bus_error *ret_error) {
+        static const struct bus_properties_map map[] = {
+                { "Interface",  "s",     map_name,       offsetof(Bearer, name)      },
+                { "Connected",  "b",     NULL,           offsetof(Bearer, connected) },
+                { "Ip4Config",  "a{sv}", map_ip4_config, 0,                          },
+                { "Ip6Config",  "a{sv}", map_ip6_config, 0,                          },
+                { "Properties", "a{sv}", map_properties, 0,                          },
+                {}
+        };
+
+        Bearer *b = ASSERT_PTR(userdata);
+        const sd_bus_error *e;
+        int r;
+
+        assert(message);
+
+        b->slot = sd_bus_slot_unref(b->slot);
+
+        e = sd_bus_message_get_error(message);
+        if (e) {
+                bool removed = false;
+
+                if (sd_bus_error_has_name(e, SD_BUS_ERROR_UNKNOWN_METHOD))
```

Comment:
hmm, this sounds like a weird error to test against here, after all this is about a disappearing object, not method, right?

it appears to me it's a misleading error code returned by mman, no?

if so, maybe also check for SD_BUS_ERROR_UNKNOWN_OBJECT here, as preparation for a future where mman returns a more appropriate error?